articles/integrate-tailwindcss-into-flask/index.md

```diff
@@ -0,0 +1,232 @@
+
+![demo1](/engineering-education/integrate-tailwindcss-into-flask/demo1.jpeg)
+
+## Installing TailwindCss
+Now that our server is up and running, let's install Tailwindcss. To install tailwind we can use either yarn or npm (depending on which one you have on your system). We  navigate to the `static` folder we created earlier from the terminal using the `cd` command, once there we'll install TailwindCss in there, as follows:
+
+```bash
+npm install tailwindcss
+```
+
+or with yarn:
+
+```bash
+yarn add tailwindcss
+```
+
+If you don't have npm you can install nodeJs since it comes with npm, or you can look at the yarn docs [here](https://classic.yarnpkg.com/en/docs/install#windows-stable) under the alternatives section. 
+Once this is done you should see that the installation includes some folders in the `static` folder, some of which include:
+
+* node_modules/ : which is where the dependencies for tailwind reside
+
+* package.json
+
+* yarn.lock (I used yarn)
+
+## Adding TailwindCss to our Flask App
+The first step is to add some of the required files in order to get tailwindcss to work. To make things easier, we'll create a new folder called `src` in the `static` folder from earlier, and in the `src` folder we'll add a file called `style.css`, and this is where tailwindCss directives will go, as we will see in a bit. To keep things clear the project folder structure should look as follows:
+
+```
+projectfolder/
+   - app.py
+   - templates/
+   - static/
+     - src/
+       - style.css
+```
+
+In this `style.css` file we'll add the following as stipulated by the tailwindcss [docs](https://tailwindcss.com/docs/installation) 
+
+```
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+```
+These are called preprocessor directives by the documentation, and you can think of them as you would preprocessor directives from a language like C++, these directives essentially paste in a lot! of utility CSS classes during compile time (A lot!). The resulting CSS from this process is what we will link to our flask templates for styling.
+
+Before we generate the CSS lets prepare a file where it should be dumped in. In the `static` folder we'll create another new folder called `css` and in here all our stylesheets will go, including the one we're about to generate. In this folder we'll create a new file called `main.css` and this is where our tailwind utility classes will go.
+ 
+### Compiling the preprocessor directives to generate CSS
+
+In order to generate, the CSS from the preprocessor directives, we'll need to issue a command from the terminal using `npx` which is a tool that comes with npm (so ensure you have npm installed, you probably already do since node is installed). Next we'll run the following command to compile and output the result into the `main.css` for our usage.
+
+```bash
+npx tailwindcss-cli@latest build ./src/style.css -o css/main.css   
```

Comment:
Try explaining what the command does. Try explaining that -o stands for output and you are basically telling tailwind where to get the prepossessed file and where to store the output. This will help the reader know how to customize the command later